packages/@aws-cdk/aws-eks/lib/manifest-resource/index.py

```diff
@@ -0,0 +1,124 @@
+import subprocess
+import os
+import json
+import logging
+import yaml
+import boto3
+from uuid import uuid4
+from botocore.vendored import requests
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+# these are coming from the kubectl layer
+os.environ['PATH'] = '/opt/kubectl:/opt/awscli:' + os.environ['PATH']
+
+outdir = os.environ.get('TEST_OUTDIR', '/tmp')
+kubeconfig = os.path.join(outdir, 'kubeconfig')
+
+CFN_SUCCESS = "SUCCESS"
+CFN_FAILED = "FAILED"
+
+def handler(event, context):
+
+    def cfn_error(message=None):
+        logger.error("| cfn_error: %s" % message)
+        cfn_send(event, context, CFN_FAILED, reason=message)
+
+    try:
+        logger.info(json.dumps(event))
+
+        request_type = event['RequestType']
+        props = event['ResourceProperties']
+        old_props = event.get('OldResourceProperties', {})
+        physical_id = event.get('PhysicalResourceId', None)
+        manifest_text = props['Manifest']
+
+        cluster_name = os.environ.get('CLUSTER_NAME', None)
+        if cluster_name is None:
+            cfn_error("CLUSTER_NAME is missing in environment")
+            return
+
+        # "log in" to the cluster
+        subprocess.check_call([ 'aws', 'eks', 'update-kubeconfig',
+            '--name', cluster_name,
+            '--kubeconfig', kubeconfig
+        ])
+
+        # convert manifest to yaml and write to file
+        manifest_yaml = yaml.dump_all(json.loads(manifest_text))
+        logger.info(manifest_yaml)
+        manifest_file = os.path.join(outdir, 'manifest.yaml')
+        with open(manifest_file, "w") as f:
+            f.write(manifest_yaml)
+        logger.info("manifest written to: %s" % manifest_file)
+
+        if request_type == 'Create' or request_type == 'Update':
+            kubectl('apply', manifest_file)
+        elif request_type == "Delete":
+            # TODO: never fail
```

Comment:
Seems like you are catching everything. Do you still need this todo? 

There are situations in which delete has challenges. I guess in general we haven't discussed what to do if pods get stuck in terminating state for example. I think kubectl returns success, but I'm not sure if you will need special error handling.

Reply:
Removed comment.